src/main/java/uk/gov/ons/ctp/response/collection/exercise/schedule/SchedulerConfiguration.java

```diff
@@ -80,11 +85,92 @@ public static boolean unscheduleEvent(final Scheduler scheduler, final Event eve
      * @return a String that can be used as a JobKey
      */
     public static String getJobKey(final UUID collexId, final Event event) {
-        String jobKey = event.getTag() + ":" + collexId.toString();
+        return getJobKey(collexId, event.getTag());
+    }
+
+    /**
+     * Utility method to generate a JobKey from a collection exercise id and an event tag
+     *
+     * @param collexId a collection exercise id
+     * @param eventTag an event tag
+     * @return a String that can be used as a JobKey
+     */
+    public static String getJobKey(final UUID collexId, final String eventTag) {
+        String jobKey = eventTag + ":" + collexId.toString();
 
         return jobKey;
     }
 
+    /**
+     * Creates a new event DTO from a quartz job
+     *
+     * @param scheduler the scheduler containing the job
+     * @param jobKey    the key for the quartz job
+     * @return a new EventDTO
+     */
+    private static EventDTO getEventDTOFromJobKey(final Scheduler scheduler, final JobKey jobKey) {
+        try {
+            EventDTO result = new EventDTO();
+            JobDetail detail = scheduler.getJobDetail(jobKey);
+            JobDataMap jobDataMap = detail.getJobDataMap();
+
+            String tagStr = jobDataMap.getString(DataKey.tag.name());
+            result.setTag(tagStr);
+
+            String collexIdStr = jobDataMap.getString(DataKey.collectionExercise.name());
+            UUID collexId = UUID.fromString(collexIdStr);
+            result.setCollectionExerciseId(collexId);
+
+            String eventIdStr = jobDataMap.getString(DataKey.id.name());
+            result.setId(UUID.fromString(eventIdStr));
+
+            Trigger trigger = scheduler.getTrigger(TriggerKey.triggerKey(getJobKey(collexId, tagStr)));
+            result.setTimestamp(trigger.getStartTime());
+
+            return result;
+        } catch (SchedulerException e) {
+            log.error("Failed to get event information for job with key {} - {}", jobKey, e.getMessage());
```

Comment:
Why not just log the stacktrace rather than just the message and should we reraise the exception to return a 500 on the end point if something is wrong with the service?

Reply:
- can log the stack trace, I am just aware that most of the overly chatty logging is probably down to me (see state machine failures and collection exercise events). 
- didn't want to re-throw the exception as it makes using the method with streams ugly but can change